src/portable/portable.c

```diff
@@ -1185,16 +1202,51 @@ int portable_attach(
                 }
         }
 
-        r = portable_extract_by_path(image->path, true, matches, NULL, &unit_files, error);
+        r = portable_extract_by_path(image->path, false, matches, ordered_hashmap_isempty(extension_images) ? NULL : &os_release, &unit_files, error);
```

Comment:
to make things simple i'd just pick os_release up always. no need to optimize here like that.